File: spring-boot-rest-integrated-demo/src/test/java/com/github/jccode/springbootrestintegrateddemo/service/UserMapperTest.java
package com.github.jccode.springbootrestintegrateddemo.service;

import com.github.jccode.springbootrestintegrateddemo.model.User;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.annotation.Rollback;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.assertThat;


@RunWith(SpringRunner.class)
@SpringBootTest
@MybatisTest
@Import(UserService.class)
public class UserMapperTest {

    @Autowired
    private UserService userService;

    @Test
    public void findUser() {
        User user = userService.find(1);
        assertThat(user.getName(), is("hello"));
        assertThat(user.getAge(), is(18));
    }

    @Test
    public void count() {
        int count = userService.count();
        assertThat(count, greaterThan(0));
    }

    @Test
    @Rollback(false)
    public void save() {
        User user = randomUser();
        int c = userService.count();

        // Insert
        int ret1 = userService.save(user);
        assertThat(ret1, greaterThan(0));
        int id = user.getId();
        assertThat(id, greaterThan(0));

        // Assert equals
        User user2 = userService.find(id);
        assertThat(user2.getName(), is(user.getName()));
        assertThat(user2.getAge(), is(user.getAge()));

        int c2 = userService.count();
        assertThat(c2, equalTo(c + 1));

        // Delete
        int ret = userService.delete(id);
        assertThat(ret, greaterThan(0));

        int c3 = userService.count();
        assertThat(c3, equalTo(c));
    }

    @Test
    @Rollback(false)
    public void updateSelective() {
        User user = randomUser();

        // Insert
        int ret1 = userService.save(user);
        assertThat(ret1, greaterThan(0));
        int id = user.getId();
        assertThat(id, greaterThan(0));

        // Update user name
        String name = String.valueOf(Math.random());
        User cond = new User();
        cond.setId(id);
        cond.setName(name);
        int ret2 = userService.updateSelective(cond);
        assertThat(ret2, greaterThan(0));

        User user2 = userService.find(id);
        assertThat(user2.getName(), is(name));
        assertThat(user2.getName(), not(is(user.getName())));
        assertThat(user2.getAge(), is(user.getAge()));

        // Delete
        userService.delete(id);
    }

    @Test
    public void findByName() {
        List<User> hello = userService.findByName("hello");
        assertThat(hello.size(), greaterThan(0));
        assertThat(hello.get(0).getName(), is("hello"));
    }

    private User randomUser() {
        String randomName = String.valueOf(Math.random());
        int age = 22;
        User user = new User();
        user.setName(randomName);
        user.setPassword("pass");
        user.setAge(age);
        return user;
    }

}

File: spring-boot-rest-integrated-demo/src/test/scala/com/github/jccode/springbootrestintegrateddemo/service/UserMapperTest.scala
package com.github.jccode.springbootrestintegrateddemo.service

import com.github.jccode.springbootrestintegrateddemo.model.User
import org.junit.Test
import org.junit.runner.RunWith
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest
import org.scalatest.Matchers
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.annotation.Rollback
import org.springframework.test.context.junit4.SpringRunner


@RunWith(classOf[SpringRunner])
@SpringBootTest
@MybatisTest
@Import(Array(classOf[UserService]))
class UserMapperTest extends Matchers {

  @Autowired
  private val userService: UserService = null

  @Test
  def findUser(): Unit = {
    val user: User = userService.find(1)
    user.getName should equal("hello")
    user.getAge should be (18)
  }

  @Test def count(): Unit = {
    val count = userService.count
    count should be > 0
  }

  @Test
  @Rollback(false)
  def save(): Unit = {
    val user = randomUser
    val c = userService.count
    // Insert
    val ret1 = userService.save(user)
    ret1 should be > 0

    val id = user.getId
    id.toInt should be > 0

    // Assert equals
    val user2 = userService.find(id)
    user2.getName should equal(user2.getName)
    user2.getAge should be(user.getAge)

    val c2 = userService.count
    c2 should be(c + 1)

    // Delete
    val ret = userService.delete(id)
    ret should be > 0

    val c3 = userService.count
    c3 should equal(c)
  }

  @Test
  @Rollback(false) def updateSelective(): Unit = {
    val user = randomUser
    val ret1 = userService.save(user)
    ret1 should be > 0

    val id = user.getId
    id.toInt should be > 0

    // Update user name
    val name = String.valueOf(Math.random)
    val cond = new User
    cond.setId(id)
    cond.setName(name)
    val ret2 = userService.updateSelective(cond)
    ret2 should be > 0

    val user2 = userService.find(id)
    user2.getName should equal(name)
    user2.getName should not equal user.getName
    user2.getAge should equal(user.getAge)

    userService.delete(id)
  }

  @Test def findByName(): Unit = {
    val hello = userService.findByName("hello")
    hello.size should be > 0
    hello(0).getName should be("hello")
  }

  private def randomUser = {
    val randomName = String.valueOf(Math.random)
    val age = 22
    val user = new User
    user.setName(randomName)
    user.setPassword("pass")
    user.setAge(age)
    user
  }

}
